src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -290,6 +294,7 @@ class PythonFaaSKnownRuntime:
     major: int
     minor: int
     tag: str
+    aws_architecture: Optional[AWSLambdaArchitecture] = None
```

Comment:
This is icky and I don't particularly love polluting common models with AWS-specific fields, but I couldn't see a better way around this

Reply:
Just brainstorming: two other options might be:

1. something like `metadata_for_matching: Optional[object]` that can be set (or not) and similarly for `RuntimePlaformsRequest`, so lose some of the benefits of typing by just treating it as an opaque `==`-able blob
2. make the architecture enum generic and move it into `faas.py`, but have the GCF targets just hard-code either `architecture=FaaSArchitecture.x86_64` or `architecture=None` (i.e. revisit https://github.com/pantsbuild/pants/pull/21234/files#r1697336354)
3. something of a middle ground: `architecture: Optional[str]` here and in the request.

Reply:
I think 3 might be best because we have no idea how GCP will denote the architecture / whether the enum values will be the same. If they do turn out to be the same then we can always tighten the string to be an enum instead.